modules/core/src/test/java/org/apache/ignite/internal/processors/performancestatistics/CheckpointTest.java

```diff
@@ -0,0 +1,273 @@
+            .setMetricsEnabled(true)
+            .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                .setMaxSize(10 * 1024 * 1024)
+                .setCheckpointPageBufferSize(1024 * 1024)
+                .setMetricsEnabled(true)
+                .setPersistenceEnabled(true))
+            .setWriteThrottlingEnabled(true)
+            .setFileIOFactory(new SlowCheckpointFileIOFactory())
+            .setCheckpointThreads(1));
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        slowCheckpointEnabled.set(false);
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testCheckpoint() throws Exception {
+        IgniteEx srv = startGrid();
+
+        srv.cluster().state(ClusterState.ACTIVE);
+
+        MetricRegistry mreg = srv.context().metric().registry(DATASTORAGE_METRIC_PREFIX);
+
+        AtomicLongMetric lastBeforeLockDuration = mreg.findMetric("LastCheckpointBeforeLockDuration");
+        AtomicLongMetric lastLockWaitDuration = mreg.findMetric("LastCheckpointLockWaitDuration");
+        AtomicLongMetric lastListenersExecDuration = mreg.findMetric("LastCheckpointListenersExecuteDuration");
+        AtomicLongMetric lastMarcDuration = mreg.findMetric("LastCheckpointMarkDuration");
+        AtomicLongMetric lastLockHoldDuration = mreg.findMetric("LastCheckpointLockHoldDuration");
+        AtomicLongMetric lastPagesWriteDuration = mreg.findMetric("LastCheckpointPagesWriteDuration");
+        AtomicLongMetric lastFsyncDuration = mreg.findMetric("LastCheckpointFsyncDuration");
+        AtomicLongMetric lastWalRecordFsyncDuration = mreg.findMetric("LastCheckpointWalRecordFsyncDuration");
+        AtomicLongMetric lastWriteEntryDuration = mreg.findMetric("LastCheckpointWriteEntryDuration");
+        AtomicLongMetric lastSplitAndSortPagesDuration =
+            mreg.findMetric("LastCheckpointSplitAndSortPagesDuration");
+        AtomicLongMetric lastDuration = mreg.findMetric("LastCheckpointDuration");
+        AtomicLongMetric lastStart = mreg.findMetric("LastCheckpointStart");
+        AtomicLongMetric lastTotalPages = mreg.findMetric("LastCheckpointTotalPagesNumber");
+        AtomicLongMetric lastDataPages = mreg.findMetric("LastCheckpointDataPagesNumber");
+        AtomicLongMetric lastCOWPages = mreg.findMetric("LastCheckpointCopiedOnWritePagesNumber");
+
+        // wait for checkpoint to finish on node start
+        assertTrue(waitForCondition(() -> 0 < lastStart.value(), TIMEOUT));
+
+        startCollectStatisticsWithImmediatelyFlush();
+
+        forceCheckpoint();
+
+        assertTrue(waitForCondition(() -> {
```

Comment:
Let's rewrite test as follows:

1. Start psproc
2. forceCheckpoint
3. Wait for checkpoint finish (based on metric value or similar)
4. Stop psproc.
5. Check is there checkpoint data in statistics (no multiple read of statistics).